src/debugger/launch.ts

```diff
@@ -328,3 +332,65 @@ function updateConfigWithNewKeys(
         oldConfig[key] = newConfig[key];
     }
 }
+
+/**
+ * Get the arguments for a launch configuration's preLaunchTask if it's a Swift build task
+ * @param launchConfig The launch configuration to check
+ * @param workspaceFolder The workspace folder context (optional)
+ * @returns Promise<string[] | undefined> the task arguments if it's a Swift build task, undefined otherwise
+ */
+export async function swiftPrelaunchBuildTaskArguments(
+    launchConfig: vscode.DebugConfiguration,
+    workspaceFolder?: vscode.WorkspaceFolder
+): Promise<string[] | undefined> {
+    const preLaunchTask = launchConfig.preLaunchTask;
+
+    if (!preLaunchTask || typeof preLaunchTask !== "string") {
+        return undefined;
+    }
+
+    try {
+        // Fetch all available tasks
+        const allTasks = await vscode.tasks.fetchTasks();
+
+        // Find the task by name
+        const task = allTasks.find(t => {
+            // Check if task name matches (with or without "swift: " prefix)
+            const taskName = t.name;
+            const matches =
+                taskName === preLaunchTask ||
+                taskName === `swift: ${preLaunchTask}` ||
+                `swift: ${taskName}` === preLaunchTask;
+
+            // If workspace folder is specified, also check scope
+            if (workspaceFolder && matches) {
+                return t.scope === workspaceFolder || t.scope === vscode.TaskScope.Workspace;
+            }
+
+            return matches;
+        });
+
+        if (!task) {
+            return undefined;
+        }
+
+        // Check if task type is "swift"
+        if (task.definition.type !== "swift") {
+            return undefined;
+        }
+
+        // Check if args contain "build" and "--build-system"
+        const args = (task.definition.args as string[]) || [];
+        const hasBuild = args.includes("build");
+        const hasBuildSystem = args.includes("--build-system");
```

Comment:
I'm not quite understanding the reason you want to check for `--build-system` here? Why do we want to take a different path for `swift build` alone vs `swift build --build-system X`?